### PR TITLE
ARROW-9222: [Format] Columnar.rst changes for removing validity bitmap from union types

### PR DESCRIPTION
**Pull request by wesm**
See mailing list discussion.

### CODE REVIEW

File: docs/source/format/Columnar.rst [hunk 1]
```diff
@@ -566,11 +572,6 @@ having the values: ``[{f=1.2}, null, {f=3.4}, {i=5}]``
...
 
     * Length: 4, Null count: 1
-    * Validity bitmap buffer:
-      |Byte 0 (validity bitmap) | Bytes 1-63            |
-      |-------------------------|-----------------------|
-      |00001101                 | 0 (padding)           |
-
     * Types buffer:
 
       |Byte 0   | Byte 1      | Byte 2   | Byte 3   | Bytes 4-63  |
```
Comment (pitrou) [inline docs/source/format/Columnar.rst:577]: It seems the types and offsets buffers need to be fixed (the second entry isn't unspecified anymore).

File: docs/source/format/Columnar.rst [hunk 2]
```diff
@@ -586,13 +587,13 @@ having the values: ``[{f=1.2}, null, {f=3.4}, {i=5}]``
...
         * Length: 2, nulls: 0
-        * Validity bitmap buffer: Not required
+        * Validity bitmap buffer: 00000101
 
         * Value Buffer:
 
-          | Bytes 0-7 | Bytes 8-63  |
-          |-----------|-------------|
-          | 1.2, 3.4  | unspecified |
+          | Bytes 0-11     | Bytes 8-63  |
```
Comment (pitrou) [inline docs/source/format/Columnar.rst:594]: Second column should be "bytes 12-63"

File: docs/source/format/Columnar.rst (hunk 1 above)
Comment (wesm) [reply docs/source/format/Columnar.rst:577]: Oops, went to fast. Fixing

File: docs/source/format/Columnar.rst (hunk 2 above)
Comment (wesm) [reply docs/source/format/Columnar.rst:594]: Fixing

File: docs/source/format/Columnar.rst [hunk 3]
```diff
@@ -688,11 +687,10 @@ will have the following layout: ::
             |------------|-----------------------|
             | joemark    | unspecified (padding) |
 
-Similar to structs, a particular child array may have a non-null slot
-even if the validity bitmap of the parent union array indicates the
-slot is null.  Additionally, a child array may have a non-null slot
-even if the types array indicates that a slot contains a different
-type at the index.
+Note that a child array may have a non-null slot even if the types array
+indicates that a slot contains a different type at the index, so the
```
Comment (emkornfield) [inline docs/source/format/Columnar.rst:691]: I have a little trouble parsing this language.  Could it be stated more as a positive?

> Only the slot in the array corresponding to the type index is considered.  All "unselected" values are ignored and could be any semantically correct array value.

File: docs/source/format/Columnar.rst [hunk 4]
```diff
@@ -566,33 +572,28 @@ having the values: ``[{f=1.2}, null, {f=3.4}, {i=5}]``
...
     * Offset buffer:
 
       |Bytes 0-3 | Bytes 4-7   | Bytes 8-11 | Bytes 12-15 | Bytes 16-63 |
       |----------|-------------|------------|-------------|-------------|
-      | 0        | unspecified | 1          | 0           | unspecified |
+      | 0        | 1           | 2          | 0           | unspecified |
 
     * Children arrays:
       * Field-0 array (f: float):
         * Length: 2, nulls: 0
```
Comment (emkornfield) [inline docs/source/format/Columnar.rst:589]: should this be nulls: 1 now?

Comment (wesm) [reply docs/source/format/Columnar.rst:589]: Yes

File: docs/source/format/Columnar.rst (hunk 3 above)
Comment (wesm) [reply docs/source/format/Columnar.rst:691]: OK

**Comment by wesm** [comment]
+1, vote result https://lists.apache.org/thread.html/r47c6578fe9c87d821ccd26e6fc28a49caa3849cca2c5e01be3c225dc%40%3Cdev.arrow.apache.org%3E

**Pull request closed without merging**